ludwig/models/predictor.py

```diff
@@ -87,6 +87,7 @@ def batch_predict(
                                     file=sys.stdout,
                                     disable=is_progressbar_disabled())
 
+            model.to(self.device)
```

Comment:
One change here could be to pass in the model during initialization of an object of `Predictor`, and move it to the correct device at initialization. Then every method of `Predictor` can use `self.model` directly, instead of moving the model to the correct device.

This should be ok for now however.

Reply:
I considered that as well, but I wasn't sure about modifying the signature of `Predictor` methods, so I went with the least intrusive approach.